safedelete/managers.py

```diff
@@ -75,63 +158,32 @@ class MyModel(models.Model):
             self._queryset_class = queryset_class
 
     def get_queryset(self):
-        # We MUST NOT do the core_filters in get_queryset.
-        # The child *RelatedManager will take care of that.
-        # It will break prefetch_related if we do it here.
+        # Backwards compatibility, no need to move options to QuerySet.
         queryset = self._queryset_class(self.model, using=self._db)
-
-        if self._safedelete_visibility in (DELETED_INVISIBLE, DELETED_VISIBLE_BY_FIELD, DELETED_ONLY_VISIBLE):
-            queryset = queryset.filter(
-                deleted__isnull=self._safedelete_visibility in (
-                    DELETED_INVISIBLE, DELETED_VISIBLE_BY_FIELD
-                )
-            )
+        queryset._safedelete_visibility = self._safedelete_visibility
+        queryset._safedelete_visibility_field = self._safedelete_visibility_field
         return queryset
 
     def all_with_deleted(self):
-        """Deprecated because all(show_deleted=True) is meant for related managers."""
-        warnings.warn('deprecated', DeprecationWarning)
```

Comment:
Might be better if we not deprecate this, but add helper methods for all visibility modifiers.